src/js/files/components/Manager.jsx

```diff
@@ -1,115 +1,132 @@
-import { capitalize, forEach } from "lodash-es";
-import React from "react";
+import { capitalize, forEach, toString } from "lodash-es";
+import React, { useEffect } from "react";
 import { connect } from "react-redux";
 import {
     Alert,
     Badge,
     Icon,
     LoadingPlaceholder,
     NoneFoundBox,
-    ScrollList,
     UploadBar,
     ViewHeader,
-    ViewHeaderTitle
+    ViewHeaderTitle,
+    Pagination
 } from "../../base";
 import { checkAdminOrPermission, createRandomString } from "../../utils/utils";
 import { findFiles, upload } from "../actions";
 import { getFilteredFileIds } from "../selectors";
 import File from "./File";
 
-export class FileManager extends React.Component {
-    componentDidMount() {
-        this.props.onLoadNextPage(this.props.fileType, this.props.term, 1);
+export const FileManager = ({
+    onLoadNextPage,
+    canUpload,
+    onDrop,
+    documents,
+    validationRegex,
+    message,
+    total_count,
+    page,
```

Comment:
Not sure it makes sense to have two sources to define the current page. I would have the url query by the single source of truth on the current page and remove page from redux.